src/pe_reports/report_generator.py

```diff
@@ -28,39 +21,221 @@
+    )
 
-def load_template():
-    """Load PowerPoint template into memory."""
-    prs = Presentation(REPORT_SHELL)
-    return prs
+    # Open csv data as binary
+    cc = open(cred_xlsx, "rb").read()
+    da = open(da_xlsx, "rb").read()
+    ma = open(vuln_xlsx, "rb").read()
+    mi = open(mi_xlsx, "rb").read()
 
+    # Insert link to csv data in summary page of pdf
+    p1 = fitz.Point(110, 695)
+    p2 = fitz.Point(240, 695)
+    p3 = fitz.Point(375, 695)
+    p5 = fitz.Point(500, 695)
 
-def export_set(prs, out_dir):
-    """Export PowerPoint report set to output directory."""
-    try:
-        prs.save(os.path.join(out_dir, REPORT_OUT))
-    except FileNotFoundError as not_found:
-        logging.error("%s : Missing input data. No report generated.", not_found)
+    # Embedd and add push-pin graphic
+    page.add_file_annot(
+        p1, cc, "compromised_credentials.xlsx", desc="Open up csv", icon="PushPin"
+    )
+    page.add_file_annot(
+        p2, da, "domain_alerts.xlsx", desc="Open up csv", icon="PushPin"
+    )
+    page.add_file_annot(p3, ma, "vuln_alerts.xlsx", desc="Open up xlsx", icon="PushPin")
+    page.add_file_annot(
+        p5, mi, "mention_incidents.xlsx", desc="Open up csv", icon="PushPin"
+    )
+
+    doc.save(
+        output,
+        garbage=4,
+        deflate=True,
+    )
+    tooLarge = False
+    # Throw error if file size is greater than 20MB
+    filesize = os.path.getsize(output)
+    if filesize >= 20000000:
+        tooLarge = True
+
+    return filesize, tooLarge
+
+
+def convert_html_to_pdf(source_html, output_filename):
+    """Convert html to pdf."""
+    # open output file for writing (truncated binary)
+    result_file = open(output_filename, "w+b")
+
+    # convert HTML to PDF
+    pisa_status = pisa.CreatePDF(
+        source_html, dest=result_file  # the HTML to convert
+    )  # file handle to recieve result
+
+    # close output file
+    result_file.close()  # close output file
+
+    # return False on success and True on errors
+    return pisa_status.err
+
+
+def generate_reports(datestring, output_directory):
+    """Process steps for generating report data."""
+    # Get PE orgs from PE db
+    conn = connect()
+    if conn:
+        pe_orgs = get_orgs(conn)
+    else:
+        return 1
+    generated_reports = 0
+
+    # Iterate over organizations
+    if pe_orgs:
+        logging.info(f"PE orgs count: {len(pe_orgs)}")
+        for org in pe_orgs:
+            # Assign organization values
+            org_uid = org[0]
+            org_name = org[1]
+            org_code = org[2]
+
+            if org_code != "DOS":
+                continue
+            logging.info(f"Running on {org_code}")
+
+            # Create folders in output directory
+            if not os.path.exists(f"{output_directory}/ppt"):
+                os.mkdir(f"{output_directory}/ppt")
+
+            if not os.path.exists(f"{output_directory}/{org_code}"):
+                os.mkdir(f"{output_directory}/{org_code}")
```

Comment:
This code can be DRY'd out a bit:
```suggestion
            for dir_name in ("ppt", org_code):
                if not os.path.exists(f"{output_directory}/{dir_name}"):
                    os.mkdir(f"{output_directory}/{dir_name}")
```

Reply:
Addressed [here](https://github.com/cisagov/pe-reports/pull/114/commits/2599135785a1ec460be4ab00e7e33db08b7f1028).